rct229/rulesets/ashrae9012019/section19/section19rule34.py

```diff
@@ -0,0 +1,111 @@
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_hvac_zone_list_w_area_dict import (
+    get_hvac_zone_list_w_area_by_rmi_dict,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_list_hvac_systems_associated_with_zone import (
+    get_list_hvac_systems_associated_with_zone,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_proposed_hvac_modeled_with_virtual_cooling import (
+    get_proposed_hvac_modeled_with_virtual_cooling,
+)
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_proposed_hvac_modeled_with_virtual_heating import (
+    get_proposed_hvac_modeled_with_virtual_heating,
+)
+from rct229.utils.assertions import getattr_
+
+
+class Section19Rule34(RuleDefinitionListIndexedBase):
+    """Rule 34 of ASHRAE 90.1-2019 Appendix G Section 19 (Envelope)"""
+
+    def __init__(self):
+        super(Section19Rule34, self).__init__(
+            rmrs_used=UserBaselineProposedVals(True, True, True),
+            each_rule=Section19Rule34.HVACRule(),
+            index_rmr="baseline",
+            id="19-34",
+            description="Where no heating and/or cooling system is to be installed, and a heating or cooling system is being simulated only to meet the requirements described in this table, "
+            "heating and/or cooling system fans shall not be simulated as running continuously during occupied hours but shall be cycled ON and OFF to meet heating and cooling loads during all hours in the baseline design.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1-4 Schedules Exception #1.",
+            is_primary_rule=False,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def create_data(self, context, data):
+        rmd_u = context.user
+        rmd_b = context.baseline
+        rmd_p = context.proposed
+
+        HVAC_systems_virtual_cooling_list_p = (
+            get_proposed_hvac_modeled_with_virtual_cooling(rmd_u, rmd_p)
+        )
+        HVAC_systems_virtual_heating_list_p = (
+            get_proposed_hvac_modeled_with_virtual_heating(rmd_u, rmd_p)
+        )
+
+        HVAC_systems_virtual_list_p = list(
+            set(
+                HVAC_systems_virtual_cooling_list_p
+                + HVAC_systems_virtual_heating_list_p
+            )
+        )
+
+        hvac_sys_zones_served_dict_p = get_hvac_zone_list_w_area_by_rmi_dict(rmd_p)
+
+        zones_virtual_heating_cooling_list = list(
+            set(
+                [
+                    hvac_sys_zones_served_dict_p[hvac_id_p]["zone_list"].value()
+                    for hvac_id_p in HVAC_systems_virtual_list_p
+                ]
+            )
+        )
+
+        applicable_hvac_with_virtual_heating_cooling_b = list(
+            set(
+                [
+                    get_list_hvac_systems_associated_with_zone(rmd_b, zone_id_p)
```

Comment:
This will also result in a 2D array.
You would need to flatten it.

Reply:
Addressed.